masonry/src/widget/kurbo.rs

```diff
@@ -0,0 +1,406 @@
+    }
+
+    pub fn set_stroke_brush(&mut self, stroke_brush: Brush) {
+        self.stroke.get_or_insert_with(Default::default).brush = stroke_brush;
+    }
+
+    pub fn set_stroke_brush_transform(&mut self, stroke_brush_transform: Option<Affine>) {
+        self.stroke
+            .get_or_insert_with(Default::default)
+            .brush_transform = stroke_brush_transform;
+    }
+}
+
+// MARK: WIDGETMUT
+impl<'a> WidgetMut<'a, KurboShape> {
+    pub fn set_shape(&mut self, shape: ConcreteShape) {
+        self.widget.shape = shape;
+        self.ctx.request_layout();
+        self.ctx.request_paint();
+        self.ctx.request_accessibility_update();
+    }
+
+    pub fn set_transform(&mut self, transform: Affine) {
+        self.widget.transform = transform;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_fill_mode(&mut self, fill_mode: Fill) {
+        self.widget.fill.get_or_insert_with(Default::default).mode = fill_mode;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_fill_brush(&mut self, fill_brush: Brush) {
+        self.widget.fill.get_or_insert_with(Default::default).brush = fill_brush;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_fill_brush_transform(&mut self, fill_brush_transform: Option<Affine>) {
+        self.widget
+            .fill
+            .get_or_insert_with(Default::default)
+            .brush_transform = fill_brush_transform;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_style(&mut self, stroke_style: Stroke) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .style = stroke_style;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_brush(&mut self, stroke_brush: Brush) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .brush = stroke_brush;
+        self.ctx.request_paint();
+    }
+
+    pub fn set_stroke_brush_transform(&mut self, stroke_brush_transform: Option<Affine>) {
+        self.widget
+            .stroke
+            .get_or_insert_with(Default::default)
+            .brush_transform = stroke_brush_transform;
+        self.ctx.request_paint();
+    }
+}
+
+// MARK: IMPL WIDGET
+impl Widget for KurboShape {
+    fn on_pointer_event(&mut self, _ctx: &mut EventCtx, _event: &PointerEvent) {}
+    fn on_text_event(&mut self, _ctx: &mut EventCtx, _event: &TextEvent) {}
+    fn on_access_event(&mut self, _ctx: &mut EventCtx, _event: &AccessEvent) {}
+    fn on_status_change(&mut self, _ctx: &mut LifeCycleCtx, _event: &StatusChange) {}
+    fn lifecycle(&mut self, _ctx: &mut LifeCycleCtx, _event: &LifeCycle) {}
+
+    fn layout(&mut self, _ctx: &mut LayoutCtx, bc: &BoxConstraints) -> Size {
+        let size = self.shape.bounding_box().size();
+        if !bc.contains(size) {
+            warn!("The shape is oversized");
+        }
+        size
+    }
```

Comment:
You probably shouldn't warn every time a widget overflows its parent.

Also, this code is valid with the Board as a parent, but I'm not sure what results it would produce with any widget type as a parent. It's not a blocking problem, though.